### DISCUSSION TITLE
Writing SVG file?

### DISCUSSION THREAD

**Discussion by frankvdh** [blackears/svgSalamander#110] [Q&A]
Hi all,
What I want to do with svgSalamander is read an SVG file, change fill colours in multiple Path elements (and maybe some other things) and write it out to a different file. I've got all of that sorted except actually writing the output.

I could iterate through all the SvgElements writing each one out in turn. Not a big deal, but rather than reinventing that wheel I'm hoping there's an easy way to do that. Something like casting the SVGRoot object as an Xml object and then calling some kind of Xml WriteFile() method? As you can see, I'm not super familiar with XML.

Frank

**Comment by blackears** [comment]
There is no exporter feature with SVGSalamander.  However, exporting it should be as simple as stepping through each node and writing all it's parameters.  The structure of the tree is the same as the XML format.
